data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/JsonSchemaConverter.java

```diff
@@ -0,0 +1,52 @@
+package org.opensearch.dataprepper.schemas;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.victools.jsonschema.generator.FieldScope;
+import com.github.victools.jsonschema.generator.Module;
+import com.github.victools.jsonschema.generator.OptionPreset;
+import com.github.victools.jsonschema.generator.SchemaGenerator;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfig;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfigBuilder;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfigPart;
+import com.github.victools.jsonschema.generator.SchemaVersion;
+
+import java.util.List;
+
+public class JsonSchemaConverter {
+    static final String DEPRECATED_SINCE_KEY = "deprecated";
+    private final List<Module> jsonSchemaGeneratorModules;
+
+    public JsonSchemaConverter(final List<Module> jsonSchemaGeneratorModules) {
+        this.jsonSchemaGeneratorModules = jsonSchemaGeneratorModules;
+    }
+
+    public ObjectNode convertIntoJsonSchema(
+            final SchemaVersion schemaVersion, final OptionPreset optionPreset, final Class<?> clazz)
+            throws JsonProcessingException {
+        final SchemaGeneratorConfigBuilder configBuilder = new SchemaGeneratorConfigBuilder(
+                schemaVersion, optionPreset);
+        loadJsonSchemaGeneratorModules(configBuilder);
+        final SchemaGeneratorConfigPart<FieldScope> scopeSchemaGeneratorConfigPart = configBuilder.forFields();
+        overrideInstanceAttributeWithDeprecated(scopeSchemaGeneratorConfigPart);
+
+        final SchemaGeneratorConfig config = configBuilder.build();
```

Comment:
Can we tell this to use snake case for generating the names?

https://github.com/opensearch-project/data-prepper/blob/7d16ea1ecf489328d61fb308f94c6811ad12e558/data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/ObjectMapperConfiguration.java#L54-L56